consensus/src/metrics.rs

```diff
@@ -0,0 +1,52 @@
+// Copyright (c) 2022, Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+use prometheus::{default_registry, register_int_gauge_vec_with_registry, IntGaugeVec, Registry};
+
+#[derive(Clone, Debug)]
+pub struct ConsensusMetrics {
+    /// The number of elements in the Dag (for Tusk or Bullshark)
+    pub consensus_dag_size: IntGaugeVec,
+    /// The last committed round from consensus
+    pub last_committed_round: IntGaugeVec,
+    /// The number of elements from the vertices secondary index (external consensus)
+    pub external_consensus_dag_vertices_elements: IntGaugeVec,
+    /// The number of elements in the dag (external consensus)
+    pub external_consensus_dag_size: IntGaugeVec,
+}
+
+impl ConsensusMetrics {
+    pub fn new(registry: &Registry) -> Self {
+        Self {
+            consensus_dag_size: register_int_gauge_vec_with_registry!(
+                "consensus_dag_size",
+                "The number of elements (certificates) in consensus dag",
+                &[],
+                registry
+            ).unwrap(),
+            last_committed_round: register_int_gauge_vec_with_registry!(
+                "last_committed_round",
+                "The most recent round that has been committed from consensus",
+                &[],
+                registry
+            ).unwrap(),
+            external_consensus_dag_vertices_elements: register_int_gauge_vec_with_registry!(
+                "external_consensus_dag_vertices_elements",
```

Comment:
Might want to prefix all Narwhal stats with `narwhal_` just for ease of discovery

Reply:
We do currently prefix all the metrics when we initialize the Registry for the primary and worker nodes https://github.com/MystenLabs/narwhal/blob/c0e057a1d2e22873238731ea8f053afdd1b01edd/node/src/metrics.rs#L14

I did it this way to ensure that those will be automatically prefixed irrespective of whether we name them that way or not . The only downside is that we we embed Narwhal in SUI and we use a common registry we lose this feature